bug: quickget --list reports grep error "grep: invalid option -- P" on macOS

**I confirm this bug has not already been reported**
- [X] I have searched the issues and this bug has not been reported previously

**Describe the bug**
`quickget --list` always reports the following error. Note it still continues and produces correct output.
```
[albert@albert-mbp quickemu]$ ./quickget --list
grep: invalid option -- P`
usage: grep [-abcdDEFGHhIiJLlMmnOopqRSsUVvwXxZz] [-A num] [-B num] [-C[num]]
	[-e pattern] [-f file] [--binary-files=value] [--color=when]
	[--context[=num]] [--directories=action] [--label] [--line-buffered]
	[--null] [pattern] [file ...]
```

**To Reproduce**
Steps to reproduce the behaviour:
1. Run `quickget` with arguments `--list`
2. See error

**System information**
  ```
----------------------------------
        Quickemu 4.9.8
----------------------------------
Distro:	macOS 15.5
Kernel:	Darwin 24.5.0 arm64
Memory:	16G
GPU:
 -Apple M2
----------------------------------
            curl 8.7.1
----------------------------------
Libraries: libcurl/8.7.1 (SecureTransport) LibreSSL/3.3.6 zlib/1.2.12 nghttp2/1.64.0
Protocols: dict file ftp ftps gopher gophers http https imap imaps ipfs ipns ldap ldaps mqtt pop3 pop3s rtsp smb smbs smtp smtps telnet tftp
Features:  alt-svc AsynchDNS GSS-API HSTS HTTP2 HTTPS-proxy IPv6 Kerberos Largefile libz MultiSSL NTLM SPNEGO SSL threadsafe UnixSockets
----------------------------------
            QEMU 10.0.3
----------------------------------
Available CPUs:
  a64fx
  arm1026
  arm1136
  arm1136-r2
  arm1176
  arm11mpcore
  arm926
  arm946
  cortex-a15
  cortex-a35
  cortex-a53
  cortex-a55
  cortex-a57
  cortex-a7
  cortex-a710
  cortex-a72
  cortex-a76
  cortex-a8
  cortex-a9
  cortex-m0
  cortex-m3
  cortex-m33
  cortex-m4
  cortex-m55
  cortex-m7
  cortex-r5
  cortex-r52
  cortex-r5f
  host
  max
  neoverse-n1
  neoverse-n2
  neoverse-v1
  pxa250 (deprecated)
  pxa255 (deprecated)
  pxa260 (deprecated)
  pxa261 (deprecated)
  pxa262 (deprecated)
  pxa270-a0 (deprecated)
  pxa270-a1 (deprecated)
  pxa270 (deprecated)
  pxa270-b0 (deprecated)
  pxa270-b1 (deprecated)
  pxa270-c0 (deprecated)
  pxa270-c5 (deprecated)
  sa1100
  sa1110
  ti925t
----------------------------------
               CPU
----------------------------------
Apple M2
  ```
**Additional context**
The error happens here:
```
function releases_pclinuxos() {
# shellcheck disable=SC2046
    echo $(web_pipe "https://ftp.fau.de/pclinuxos/pclinuxos/iso" | grep -oP 'pclinuxos64-\K[^\-]+-\K[0-9]+\.[0-9]+' | head -n 1)
}
```
The option `-P` is only available with GNU grep, but macOS comes with BSD grep:

```
[albert@albert-mbp quickemu]$ grep -V
grep (BSD grep, GNU compatible) 2.6.0-FreeBSD
```

--- a/quickget
+++ b/quickget
@@ -929,7 +929,7 @@ function editions_parrotsec() {
 
 function releases_pclinuxos() {
 # shellcheck disable=SC2046
-    echo $(web_pipe "https://ftp.fau.de/pclinuxos/pclinuxos/iso" | grep -oP 'pclinuxos64-\K[^\-]+-\K[0-9]+\.[0-9]+' | head -n 1)
+    echo $(web_pipe "https://ftp.fau.de/pclinuxos/pclinuxos/iso" | grep -o -E 'pclinuxos64-([a-z]+-)+[0-9]{4}\.[0-9]{2}\.iso'| grep -o -E '[0-9]{4}\.[0-9]{2}'| head -n 1)
 }
 
 function editions_pclinuxos() {

